interfaces/builtin/usb_raw.go

```diff
@@ -0,0 +1,45 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const usbrawConnectedPlugAppArmor = `
+# Description: Allow raw access to all connected USB devices.
+# Reserved because this gives privileged access to the system.
+# Usage: reserved
+/dev/bus/usb/[0-9][0-9][0-9]/[0-9][0-9][0-9] rw,
+
+# Allow detection of usb devices. Leaks plugged in USB device info
+/sys/bus/usb/devices/ r,
+/sys/devices/pci**/usb[0-9]**/idVendor r,
+￼/sys/devices/pci**/usb[0-9]**/idProduct r,
+`
```

Comment:
why do you access data via /sys/bus/pci**/usb** instead of /sys/bus/usb**/?
it seems that the information is in both tree
```
$ ll /sys/bus/usb/devices/*/*/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:59 /sys/bus/usb/devices/1-11/1-11:1.0/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:59 /sys/bus/usb/devices/1-4/1-4:1.0/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:59 /sys/bus/usb/devices/1-8/1-8:1.0/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:59 /sys/bus/usb/devices/1-8/1-8:1.1/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:59 /sys/bus/usb/devices/1-8/1-8:1.2/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:59 /sys/bus/usb/devices/1-8/1-8:1.3/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:59 /sys/bus/usb/devices/2-2/2-2:1.0/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:59 /sys/bus/usb/devices/2-4/2-4:1.0/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:59 /sys/bus/usb/devices/3-1:1.0/0003:046D:C404.0001/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:59 /sys/bus/usb/devices/3-1/3-1:1.0/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:58 /sys/bus/usb/devices/usb1/1-0:1.0/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:58 /sys/bus/usb/devices/usb2/2-0:1.0/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:58 /sys/bus/usb/devices/usb3/3-0:1.0/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:58 /sys/bus/usb/devices/usb4/4-0:1.0/modalias
$ ll /sys/bus/pci/devices/*/usb*/*/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:58 /sys/bus/pci/devices/0000:00:14.0/usb1/1-0:1.0/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:58 /sys/bus/pci/devices/0000:00:14.0/usb2/2-0:1.0/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:58 /sys/bus/pci/devices/0000:04:00.0/usb3/3-0:1.0/modalias
-r--r--r-- 1 root root 4,0K nov  2 10:58 /sys/bus/pci/devices/0000:04:00.0/usb4/4-0:1.0/modalias
```